docs/docs/migrating-from-v1-to-v2.md

```diff
@@ -175,22 +179,203 @@ export default props => (
+
+Note: there will be a `postcss` plugin that allows you to configure postcss from a standard postcss config file. [Follow this discussion on issue 3284](https://github.com/gatsbyjs/gatsby/issues/3284).
+
+## Files mixing CommonJS with ES6 modules won't compile
+
+Gatsby v2 uses babel 7 which is stricter about parsing files with mixed JS styles.
+
+ES6 Modules are ok:
+
+```js
+import foo from "foo"
+export default foo
+```
+
+CommonJS is ok:
+
+```js
+const foo = require('foo');
+module.exports = foo;
+```
+
+Mixing `requires` and `export` is not ok:
+```js
+const foo = require('foo');
+export default foo
+```
+
+Mixing `import` and `module.exports` is not ok:
+```js
+import foo from "foo"
+module.exports = foo;
+```
+
+See [Gatsby's babel docs for more details](/docs/babel).
+
+## Don't query nodes by ID
+
+Source and transformer plugins now use UUIDs for IDs. If you used glob or regex to query nodes by id then you'll need to query something else.
+
+[See the Pull Request that implemented this change](https://github.com/gatsbyjs/gatsby/pull/3807/files)
+
+> TODO: add example
+
+## Remove explicit polyfills
+
+If your Gatsby v1 site included any polyfills, you can remove them. Gatsby v2 ships with babel 7 and is configured to automatically include polyfills for your code. See [Gatsby's babel docs for more details](/docs/babel).
+
+## Change `modifyBabelrc` to `onCreateBabelConfig`
+
+`modifyBabelrc` was renamed to [`onCreateBabelConfig`](/docs/node-apis/#modifyBabelrc) to bring it in line with the rest of Gatsby's API names.
+
+Before:
+
+```js
+exports.modifyBabelrc = ({ babelrc }) => {
+  return {
+    ...babelrc,
+    plugins: babelrc.plugins.concat([`foo`]),
+  }
+}
+```
+
+After:
+
+```js
+exports.onCreateBabelConfig = ({ actions }) => {
+  actions.setBabelPlugin({
+    name: `babel-plugin-foo`,
+  })
+}
+```
+
+Note usage of the new [`setBabelPlugin` action](/docs/actions/#setBabelPlugins).
+
+See [Gatsby's babel docs for more details](/docs/babel) about configuring babel.
+
+## Change `modifyWebpackConfig` to `onCreateWebpackConfig`
+
+`modifyWebpackConfig` was renamed to [`onCreateWebpackConfig`](/docs/node-apis/#onCreateWebpackConfig) to bring it in line with the rest of Gatsby's API names.
+
+Before:
+
+```js
+exports.modifyWebpackConfig = ({ config, stage }) => {
+  switch (stage) {
+    case `build-javascript`:
+      config.plugin(`Foo`, webpackFooPlugin, null)
+      break
+  }
+  return config
+}
+```
+
+
+After:
+
+```js
+exports.onCreateWebpackConfig = ({ stage, actions }) => {
+  switch (stage) {
+    case `build-javascript`:
+      actions.setWebpackConfig({
+        plugins: [webpackFooPlugin],
+      })
+  }
+}
+```
+
+Note usage of the new [`setWebpackConfig` action](/docs/actions/#setWebpackConfig).
+
+See [Gatsby's webpack docs for more details](/docs/add-custom-webpack-config) about configuring webpack.
+
+## CSS is inlined automatically
```

Comment:
Perhaps "Remove inlined css in html.js"?

Reply:
👍